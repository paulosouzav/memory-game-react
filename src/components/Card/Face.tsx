import React from 'react';
import styled, { css } from 'styled-components';
import { Image } from '../Image';
import { ROTATION } from '../../constants/rotation';

interface Props {
  front?: boolean;
  matched?: boolean;
}

const Face = styled(({ front, matched, ...props }) => <Image {...props} />)`
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;

  transition: opacity 0.4s;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;

  ${({ front }: Props) => front && ROTATION.half}
  ${({ matched }: Props) => matched && MATCHED}
`;

const MATCHED = css`
  opacity: 0.5;
`;

Face.displayName = 'Face';

export default Face;
