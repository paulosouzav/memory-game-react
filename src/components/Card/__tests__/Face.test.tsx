import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import Face from '../Face';

describe('Face', () => {
  it('should render without failing', () => {
    const wrapper = mount(<Face />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should only pass valid props to Image', () => {
    const src = 'fake/path/img.png';
    const props = {
      front: true,
      matched: true,
    };
    const wrapper = mount(<Face {...{ ...props, src }} />);
    const image = wrapper.find('Image');
    expect(image.props()).toEqual(expect.not.objectContaining(props));
  });
});
