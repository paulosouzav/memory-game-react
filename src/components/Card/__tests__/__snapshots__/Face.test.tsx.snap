// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Face should render without failing 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;
  -webkit-transition: opacity 0.4s;
  transition: opacity 0.4s;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
}

<Face>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;

  transition: opacity 0.4s;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;

  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Face",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
  >
    <Component
      className="c0"
    >
      <Image
        className="c0"
      >
        <img
          className="c0"
        />
      </Image>
    </Component>
  </StyledComponent>
</Face>
`;
